pages/docs/react/latest/hooks-overview.mdx

```diff
@@ -0,0 +1,112 @@
+  };
+
+  let msg = "You clicked" ++ Belt.Int.toString(count) ++  "times"
+
+  <div>
+    <p>{React.string(msg)}</p>
+    <button onClick> {React.string("Click me")} </button>
+  </div>
+}
+```
+```js
+function Counter(Props) {
+  var match = React.useState(function () {
+        return 0;
+      });
+  var setCount = match[1];
+  var onClick = function (_evt) {
+    return Curry._1(setCount, (function (prev) {
+                  return prev + 1 | 0;
+                }));
+  };
+  var msg = "You clicked" + String(match[0]) + "times";
+  return React.createElement("div", undefined, React.createElement("p", undefined, msg), React.createElement("button", {
+                  onClick: onClick
+                }, "Click me"));
+}
+```
+
+</CodeTab>
+
+
+Here we are using the `React.useState` Hook. We call it inside a component function to add some local state to it. React will preserve this state between re-renders. `React.useState` returns a tuple: the current state value (`count`) and a function that lets you update it (`setCount`). You can call this function from an event handler or pass it down to other components to call the function. 
+
+The only argument to `React.useState` is a function that returns the initial state (`_ => 0`). In the example above, it is 0 because our counter starts from zero. Note that your state can be any type you want and `ReScript` will make sure to infer the types for you (only make sure to return an initial state that matches your type). The initial state argument is only used during the first render.
+
+This was just a quick example on our first hook usage. We will go into more detail in a dedicated [useState](./hooks-state) section.
+
+## Available Hooks
+
+**Note:** All hooks are part of the `React` module (e.g. `React.useState`).
+
+### Basic Hooks:
+
+- [useState](./hooks-state): Adds local state to your component
+- [useEffect](./hooks-effect): Runs side-effectual code within your component
+- [useContext](./hooks-context): Gives your component to a React Context value
+
+### Additional Hooks:
+
+- [useReducer](./hooks-reducer): An alternative to `useState`. Uses the state / action / reduce pattern.
+<!-- - [useCallback](./hooks-callback): Returns a memoized callback -->
+<!-- - [useMemo](./hooks-memo): Returns a memoized value -->
+- [useRef](./hooks-ref): Returns a mutable React-Ref value
+<!-- - [useImperativeHandle](./hooks-imperative-handle): Customizes the instance value that is exposed to parent components when using `ref` -->
+<!-- - [useLayoutEffect](./hooks-layout-effect): Identical to useEffect, but it fires synchronously after all DOM mutations. -->
+
+
+## Rules of Hooks
+
+Hooks are just simple functions, but you need to follow _two rules_ when using them. ReScript doesn't enforce those rules within the compiler, so if you really want to enforce correct hooks conventions, you can use an [eslint-plugin](https://www.npmjs.com/package/eslint-plugin-react-hooks) to check your compiled JS output.
```

Comment:
I'm just doing this (eslint the compiled JS output) and wondering whether it's the right way!